README.md

```diff
@@ -78,6 +107,106 @@ Properties:
+| user    |            String or nil            |    No    | Owner of the current Splunk installation, if any | User that will be used to write to the .conf files.                                                                                                                                          |
+| reset   |            true or false            |    No    | false                                            | When specified as true, entirely replaces existing config. By default, config is merged into the existing conf file.                                                                         |
+
+##### Configuration
+The configuration you provide will be evaluated against the existing config file (if it exists, and if reset is not specified).
+Symbol or string keys are irrelevant as the provided config is converted to string keys and values before being used.
+
+Some things to be aware of when using the configuration resource:
+
+###### **"Global" properties are evaluated under `[default]`**
+Properties defined at the top of the .conf file and not considered to be part of a stanza are assumed to be `[default]`
+per Splunk documentation: http://docs.splunk.com/Documentation/Splunk/6.4.2/Admin/Propsconf#GLOBAL_SETTINGS
+
+###### **Comments are not preserved**
+The configuration resource can't evaluate and merge the configuration on disk and by Chef while keeping the
+relevant comments and whitespace. The .conf file will be rewritten with the merged config and uniform whitespace.
+
+###### **Example of how the configuration is written out to disk:**
+
+```Ruby
+splunk_conf 'system/test.conf' do
+  config(
+    testing: {
+      one: 1,
+      two: 2,
+      three: 3
+    },
+    stanza: {
+      key: :value
+    }
+  )
+end
+```
+
+`/opt/splunk/etc/system/local/test.conf`:
+```INI
+# Warning: This file is managed by Chef!
+# Comments will not be preserved and configuration may be overwritten.
+
+[testing]
+one = 1
+two = 2
+three = 3
+
+[stanza]
+key = value
+```
+
+##### Run State
+Since there is no concept of type in .conf files, everything is evaluated as strings. Configuration is read as strings
+and the provided config is converted to strings before merging. `splunk_conf` will store config (after merging) into the
+run state when its configure action is run, attempting to convert boolean and number types from strings for convenience.
+
+Keep in mind when accessing the run state:
+* All config keys will be strings regardless of whether you used strings or symbols
+* The `:config` key will not exist in the installation run state until `splunk_conf` is run
+
+```Ruby
+node.run_state['splunk_ingredient'] = {
+  'current_installation' => {
+    name: 'splunk',
+    package: :splunk,
+    version: '5.5.5',
+    build: 'a1b2c3d4e5f6',
+    x64: true,
+    config: {
+      'testing' => {
```

Comment:
This example isn't accurate.  It should have the path as a parent key to the configs contained in that path.